test/extensions/transport_sockets/tls/integration/ssl_integration_test.cc

```diff
@@ -87,6 +88,57 @@ void SslIntegrationTestBase::checkStats() {
   counter->reset();
 }
 
+class SslKeyLogTest : public SslIntegrationTest {
+public:
+  SslKeyLogTest() { config_helper_.addRuntimeOverride("tls_keylog", "true"); };
+  void setLocalFilter() { debug_tls_keylog_mode_ = 0x1; }
+  void setRemoteFilter() { debug_tls_keylog_mode_ = 0x10; }
+  void setBothLocalAndRemoteFilter() { debug_tls_keylog_mode_ = 0x11; }
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, SslKeyLogTest,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+TEST_P(SslKeyLogTest, SetLocalFilter) {
```

Comment:
Please add a test for correct content of the keylog file. You can probably use some existing utilities for access logs, something similar to `EXPECT_THAT(waitForAccessLog(access_log_name_), HasSubstr("HTTP/1.0"));` (which I copied from integration_test.cc).